### DISCUSSION TITLE
How to get a list of classes via package name within Quarkus native image?

### DISCUSSION THREAD

**Discussion by dhoffer** [quarkusio/quarkus#42339] [Q&A]
We have Java code that needs to find classes from a specified package name and then read their annotations. We have tried this using org.reflections, Guava's ClassLoader & with the JDK's Thread.currentThread().getContextClassLoader() approach and all three of these work when run on the JVM.

However when we compile the same application as a native Quarkus app using GraalVM or Mandrel all three approaches do not work, none of the classes are found that exist in the specified package name.

We have added the @RegisterForReflection annotations on all the classes we expect to find in this package.

How can I get this to work in a native GraalVM or Mandrel application? Note I am currently building the native for Windows.

We have tried this using org.reflections, Guava's ClassLoader & with the JDK's Thread.currentThread().getContextClassLoader() approach and we expect any of these should allow us to find classes that have a certain package name. All these work when run on the JVM but none work when the app is natively compiled by GraalVM or Mandrel.

[a comment by an automated account is not shown]

**Comment by Karm** [comment]
Hello @dhoffer, put together a small program, preferably a Quarkus maven project we can take a look at, please.

**Comment by dhoffer** [comment]
That probably won't be possible, the small program probably won't have this
issue.  I can say that this particular app has several dependencies and the
classes that it can't find come from one of those dependencies.  Yes, all
of our dependencies have an empty beans.xml file andor a Jandex index
file.  Everything works as expected when run with the JVM.   Also all have
@RegisterForReflection.

**Reply by Karm** [reply]
@dhoffer I wouldn't go as far as call it an issue. You are trying to do something that must have a specific config in the context of Native image. If you preapre a small Quarkus project that tries to do what you'd like, on an arbitrary class from an arbitrary third party dependency, I will take a look and make it work or explain why it doesn't. It could be your own artificial one, dhoffer.hello.world.MyClass.  Without having something to look at, we are going to burn a lot of time trying to understand each other...

**Comment by dhoffer** [comment]
I'll see if I can recreate with a new simple app.
